controllers/volumereplication_controller.go

```diff
@@ -135,7 +135,7 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		volumeHandle = pv.Spec.CSI.VolumeHandle
 	default:
 		_ = r.updateReplicationStatus(instance, replicationv1alpha1.ReplicationFailure, ("unsupported datasource kind " + instance.Spec.DataSource.Kind))
-		return ctrl.Result{}, fmt.Errorf("unsupported datasource kind %q", instance.Spec.DataSource.Kind)
+		return ctrl.Result{Requeue: false}, fmt.Errorf("unsupported datasource kind %q", instance.Spec.DataSource.Kind)
```

Comment:
does this change show the error in the logs for the user?

Reply:
It should do something like 2nd error from `controller-runtime.manager.controller.volumereplication`. But I have not tested 
it out. Anyway, I have made changes.

```
2021-03-26T08:25:28.059Z	ERROR	controllers.VolumeReplication	Failed to Replicate	{"Name": "vr1", "ReplicationState": "Resyncing", "error": "unsupported image state \"Resyncing\""}


2021-03-26T08:25:28.069Z	ERROR	controller-runtime.manager.controller.volumereplication	Reconciler error	{"reconciler group": "replication.storage.openshift.io", "reconciler kind": "VolumeReplication", "name": "vr1", "namespace": "rook-ceph", "error": "unsupported image state \"Resyncing\""}
```

Reply:
if not reconciling, then I would suggest just logging the error and returning nil.

Reply:
Done